CMakeLists.txt

```
@@ -1,198 +1,216 @@
-  macro(qt_wrap_ui)
-    qt5_wrap_ui(${ARGN})
-  endmacro()
-endif()
-## Prefer the Qt signals and slots to avoid defining "emit", "slots",
-## etc because they can conflict with boost signals, so define QT_NO_KEYWORDS here.
+find_package(tf2 REQUIRED)
+find_package(tf2_eigen REQUIRED)
+find_package(tf2_geometry_msgs REQUIRED)
+
+find_package(interactive_markers REQUIRED)
+
+find_package(Qt5 REQUIRED COMPONENTS Widgets)
+
+## Qt5 boilerplate options from http://doc.qt.io/qt-5/cmake-manual.html
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 set(CMAKE_AUTOMOC ON)
+
+## Prefer the Qt signals and slots to avoid defining "emit", "slots",
+## etc because they can conflict with boost signals, so define QT_NO_KEYWORDS here.
 set(CMAKE_AUTOMOC_COMPILER_PREDEFINES OFF)
 add_definitions(-DQT_NO_KEYWORDS)
 
-catkin_package(
-  LIBRARIES
-    ${PROJECT_NAME}
-    ${PROJECT_NAME}_gui
-    ${PROJECT_NAME}_remote_control
-    ${PROJECT_NAME}_imarker_simple
-  CATKIN_DEPENDS
-    eigen_conversions
-    geometry_msgs
-    graph_msgs
-    shape_msgs
-    roscpp
-    sensor_msgs
-    std_msgs
-    tf_conversions
-    trajectory_msgs
-    visualization_msgs
-  INCLUDE_DIRS
-    include
-    ${OGRE_OV_INCLUDE_DIRS}
-)
-
-include_directories(SYSTEM include ${catkin_INCLUDE_DIRS} ${EIGEN3_INCLUDE_DIRS} ${OGRE_OV_INCLUDE_DIRS})
-
 ###########
 ## Build ##
 ###########
-
-set(SOURCE_FILES
-  src/${PROJECT_NAME}_gui.cpp
-  src/key_tool.cpp
-)
+include_directories(
+  include
+  ${Eigen3_INCLUDE_DIRS})
 
 # Rviz GUI library
-add_library(${PROJECT_NAME}_gui ${SOURCE_FILES} src/remote_control.cpp)
-target_link_libraries(${PROJECT_NAME}_gui ${rviz_DEFAULT_PLUGIN_LIBRARIES} ${QT_LIBRARIES} ${catkin_LIBRARIES})
+# set(SOURCE_FILES
+  # src/${PROJECT_NAME}_gui.cpp
+  # src/key_tool.cpp
+# )
+# add_library(${PROJECT_NAME}_gui ${SOURCE_FILES} src/remote_control.cpp)
+# target_include_directories(${PROJECT_NAME}_gui PUBLIC
+  # ${OGRE_INCLUDE_DIRS}
+# )
+# target_link_libraries(${PROJECT_NAME}_gui Qt5::Widgets)
+# ament_target_dependencies(${PROJECT_NAME}_gui
+  # rclcpp
+# )
 
 # Remote control library
-add_library(${PROJECT_NAME}_remote_control src/remote_control.cpp)
-target_link_libraries(${PROJECT_NAME}_remote_control ${catkin_LIBRARIES})
+add_library(${PROJECT_NAME}_remote_control
+  src/remote_control.cpp
+)
+target_link_libraries(${PROJECT_NAME}_remote_control)
+ament_target_dependencies(${PROJECT_NAME}_remote_control
+  rclcpp
+  sensor_msgs
+)
 
 # Visualization Tools Library
-add_library(${PROJECT_NAME}
-  src/${PROJECT_NAME}.cpp
+add_library(${PROJECT_NAME} SHARED
+  src/rviz_visual_tools.cpp
   src/tf_visual_tools.cpp
+  src/remote_control.cpp
 )
-target_link_libraries(${PROJECT_NAME}
-  ${PROJECT_NAME}_remote_control
-  ${catkin_LIBRARIES}
-  ${Boost_LIBRARIES}
+
+target_link_libraries(${PROJECT_NAME})
+
+ament_target_dependencies(${PROJECT_NAME}
+  rclcpp
+  rclcpp_components
+  visualization_msgs
+  tf2
+  tf2_eigen
+  tf2_geometry_msgs
+  sensor_msgs
+  std_msgs
```

```suggestion
  shape_msgs
  std_msgs
  trajectory_msgs
```

Need these guys - already in find_package and export_dep, just not here